src/test/java/com/google/ehub/servlets/DashboardServletTest.java

```diff
@@ -0,0 +1,129 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.ehub.data.EntertainmentItemDatastore;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class DashboardServletTest {
+  private final DashboardServlet servlet = new DashboardServlet();
+  private final PrintWriter printWriter = new PrintWriter(new StringWriter());
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private static final String SEARCH_VALUE_PARAMETER_KEY = "searchValue";
+  private static final String SORTING_DIRECTION_PARAMETER_KEY = "sortingDirection";
+  private static final String ASCENDING_PARAMETER_VALUE = "ASCENDING";
+  private static final String JSON_CONTENT_TYPE = "application/json";
+
+  private static final int MAX_SEARCH_VALUE_CHARS = 150;
+
+  @Mock HttpServletRequest request;
+
+  @Mock HttpServletResponse response;
+
+  @Before
+  public void init() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void getRequestWithNullParams_NoContentGetsSent() throws IOException {
+    when(request.getParameter(SEARCH_VALUE_PARAMETER_KEY)).thenReturn(null);
+    when(request.getParameter(SORTING_DIRECTION_PARAMETER_KEY)).thenReturn(null);
+    when(response.getWriter()).thenReturn(printWriter);
+
+    servlet.doGet(request, response);
+
+    verify(response, never()).setContentType(JSON_CONTENT_TYPE);
+  }
+
+  @Test
+  public void getRequestWithEmptyParams_NoContentGetsSent() throws IOException {
+    when(request.getParameter(SEARCH_VALUE_PARAMETER_KEY)).thenReturn("");
+    when(request.getParameter(SORTING_DIRECTION_PARAMETER_KEY)).thenReturn("");
+    when(response.getWriter()).thenReturn(printWriter);
+
+    servlet.doGet(request, response);
+
+    verify(response, never()).setContentType(JSON_CONTENT_TYPE);
+  }
+
+  @Test
+  public void getRequestWithValidParams_ContentGetsSent() throws IOException {
+    when(request.getParameter(SEARCH_VALUE_PARAMETER_KEY)).thenReturn("Star Wars");
+    when(request.getParameter(SORTING_DIRECTION_PARAMETER_KEY))
+        .thenReturn(ASCENDING_PARAMETER_VALUE);
+    when(response.getWriter()).thenReturn(printWriter);
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType(JSON_CONTENT_TYPE);
```

Comment:
In this test case you can also verify that the response contains json for the entertainment items. 

You can add a single item to the datastore and verify it is part of the response.

Reply:
I added checks to see if the response sends the correct Json object for entertainment items in last commit.